File: OutInEvent.java
/**
 * Created by veraxmedax on 04/06/2017.
 */
public class OutInEvent extends Event {

    private OutputDevice outputDevice;

    public OutInEvent(Simulation sim, Task task, double time, OutputDevice outputDevice) {
        super(sim, task, time);
        this.outputDevice = outputDevice;
    }

    public OutputDevice getOutputDevice() {
        return outputDevice;
    }

    public void setOutputDevice(OutputDevice outputDevice) {
        this.outputDevice = outputDevice;
    }

    @Override
    void process() {
        getOutputDevice().work(getTask(), getTime());
    }
}

File: OutInEvent.cpp
//
// Created by Can Candan on 04/06/17.
//

#include "OutInEvent.h"

OutInEvent::OutInEvent(Simulation *pSimulation, Task *pTask, double time, OutputDevice *pDevice) : Event(pSimulation, pTask, time) {
    this->out=pDevice;
    name="OutInEvent";
}

void OutInEvent::process() {
    log();
    out->work(task, time);
}
